common/src/main/java/org/opensearch/sdk/SdkClient.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.sdk;
+
+import java.util.concurrent.CompletionStage;
+
+public class SdkClient {
```

Comment:
This should be an interface and XContentClient should implement the the interface.

Reply:
Yes, I know and if you look at the commit history you'll see it was initially implemented that way.

The problem I encountered is trying to inject it into the transport actions via `@Inject` annotation, which means I needed to return an `SdkClient` instance via the `createComponents()` extention point.   OpenSearch just uses the runtime class as the key to bind to, which is the concrete implementation, not the class.   It's possible to bind to the interface if you have access to the injector, but plugins don't.  

Eventually this object/implementation will probably be provided to the plugin at implementation, and this whole problem will go away.

I could create a new wrapper class to inject and still build against the interface, which will look a lot like this class.  In the interest of moving forward with the intent to get multi-tenancy solved within the next week, I don't want to spend too much time on this temporary workaround.

Reply:
Plugins can make use `createGuiceModules` to inject custom guice modules. In the custom guice module we should be able to inject required implementation. Sample code to override plugin module: https://github.com/arjunkumargiri/ml-commons/commit/49e8f1ad95922f61e6be8b806a1808179f0c6c79#diff-5fb7a6291568399efed0b4be65f70e91206634d3a1e5d1265243f1c204c0d81f

Reply:
Nice! I had forgotten that extension point existed!  I'll go ahead and use it.